frontend/hub/collections/hooks/useCopyToRepository.tsx

```diff
@@ -0,0 +1,245 @@
+import { CollectionVersionSearch } from '../Collection';
+import { usePageDialog } from './../../../../framework';
+import { Button, Modal, ModalVariant } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useState, useEffect, useCallback } from 'react';
+import {
+  useRepositoryColumns,
+  useRepositoryFilters,
+} from './../../repositories/hooks/useRepositorySelector';
+import { PageTable } from './../../../../framework/PageTable/PageTable';
+import { usePulpView } from '../../usePulpView';
+import { AnsibleAnsibleRepositoryResponse } from './../../api-schemas/generated/AnsibleAnsibleRepositoryResponse';
+import { pulpAPI, hubAPI, hubAPIPost } from '../../api/utils';
+import { useGetRequest } from './../../../common/crud/useGet';
+import { HubItemsResponse } from '../../useHubView';
+import { PulpItemsResponse } from '../../usePulpView';
+import { parsePulpIDFromURL } from '../../api/utils';
+import { useHubContext, HubContext } from './../../useHubContext';
+import { SigningServiceResponse } from '../../api-schemas/generated/SigningServiceResponse';
+import { AwxError } from '../../../awx/common/AwxError';
+
+export function useCopyToRepository() {
+  const [_, setDialog] = usePageDialog();
+
+  const context = useHubContext();
+  const onClose = useCallback(() => setDialog(undefined), [setDialog]);
+
+  return (collection: CollectionVersionSearch, operation: 'approve' | 'copy') => {
+    setDialog(
+      <CopyToRepositoryModal
+        collection={collection}
+        context={context}
+        onClose={onClose}
+        operation={operation}
+      />
+    );
+  };
+}
+
+function CopyToRepositoryModal(props: {
```

Comment:
This works well for now.
 
I believe we can also utilize the `MultiSelectDialog` framework component here to create the modal for selecting the repositories. We may need to expand it to accept a "disableRow" prop that allows disabling selection on existing repositories.

Reply:
Yes, as I have wrote this in description, this should be separate PR.